lib/tests/tests/diagnostics/unhandled_parse_regex_all_type.vrl

```diff
@@ -9,10 +9,12 @@
 #   │ │    │
 #   │ │    this expression is fallible
 #   │ │    update the expression to be infallible
+#   │ │    note if an argument type is invalid it can render a function fallible
```

Comment:
I think it would be great if we could go even further and say, for cases where the function is fallible due to the argument type not being known to be valid, that:

```
this expression is fallible because argument `.result[0].an` is of type `any` and this function expects a `string`. It will fail at runtime if `.result[0].an` is not a `string`
```

The currently added message is definitely an improvement, but I think users will remain largely confused by this behavior of VRL.

Reply:
@jszwedko: Agreed, good point. 

However, I don't have time to do it before this upcoming release. Two options:

1. Get this improvement in and follow up with another PR (to be included in the next release)
2. Leave this PR open

What do you prefer?

Reply:
👍 I'm fine with merging this since it is an improvement and then following up. I just want to make sure we don't close out https://github.com/vectordotdev/vrl/issues/23 until we do that.

Reply:
Sounds good. Added a reminder [here](https://datadoghq.atlassian.net/browse/OPW-89?focusedCommentId=1390837). 

Merging this PR won't close the ticket, so if you give this a ✅ I will include it in the VRL release.